third-party/re2/README

```diff
@@ -2,7 +2,7 @@
 re2 README for Chapel
 =====================
 
-This copy of re2 2017-04-01 (2b16c27) is being released with Chapel
+This copy of re2 2017-07-01 (a810d71) is being released with Chapel
```

Comment:
Ok. I don't think anybody but us will ever read these READMEs anyways, but I changed it again